File: labs/Week 3 - Spring Framework and Enterprise Patterns/module-26/lab26/lab26-crm/src/main/java/com/northstar/crm/entity/Customer.java
package com.northstar.crm.entity;

public class Customer {
    private String customerId;
    private String fullName;
    private String email;
    private CustomerStatus status;

    public Customer() {}

    public Customer(String customerId, String fullName, String email, CustomerStatus status) {
        this.customerId = customerId;
        this.fullName = fullName;
        this.email = email;
        this.status = status;
    }

    public static Customer amina() {
        return new Customer("CUS-1001", "Amina Khan", "amina.khan@example.com", CustomerStatus.ACTIVE);
    }

    public static Customer ravi() {
        return new Customer("CUS-1002", "Ravi Singh", "ravi.singh@example.com", CustomerStatus.PROSPECT);
    }

    public String getCustomerId() { return customerId; }
    public void setCustomerId(String customerId) { this.customerId = customerId; }
    public String getFullName() { return fullName; }
    public void setFullName(String fullName) { this.fullName = fullName; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    public CustomerStatus getStatus() { return status; }
    public void setStatus(CustomerStatus status) { this.status = status; }
}

File: labs/Week 4 - Kafka, React, PostgreSQL and Resilience/module-35/lab35/lab35-crm/crm-ui/src/types/customer.ts
export type CustomerStatus = 'PROSPECT' | 'ACTIVE' | 'SUSPENDED' | 'CLOSED'

export interface Customer {
  customerId: string
  fullName: string
  email: string
  phone: string
  status: CustomerStatus
}

export type CustomerDraft = Omit<Customer, 'customerId'>

export type Mode =
    | { kind: 'closed' }
    | { kind: 'create' }
    | { kind: 'edit'; id: string }